executor/src/witgen/identity_processor.rs

```diff
@@ -113,6 +116,29 @@ impl<'a, 'b, T: FieldElement> IdentityProcessor<'a, 'b, T> {
         unimplemented!("No executor machine matched identity `{identity}`")
     }
 
+    pub fn split_machines<'c>(
+        machines: &'c mut [&'b mut KnownMachine<'a, T>],
+        i: usize,
+    ) -> (
+        &'c mut KnownMachine<'a, T>,
+        Vec<&'c mut KnownMachine<'a, T>>,
```

Comment:
This is weird, I'd actually expect to get lifetime `'b` instead of `'c`, but the compiler won't let me.

Reply:
I think if we end up having 3 lifetimes there must be something we're doing wrong